seqr/views/apis/dataset_api.py

```diff
@@ -35,75 +35,77 @@ def add_variants_dataset_handler(request, project_guid):
     project = get_project_and_check_permissions(project_guid, request.user, can_edit=True)
     request_json = json.loads(request.body)
 
-    try:
-        required_fields = ['elasticsearchIndex', 'datasetType']
-        if any(field not in request_json for field in required_fields):
-            raise ValueError('request must contain fields: {}'.format(', '.join(required_fields)))
-        elasticsearch_index = request_json['elasticsearchIndex'].strip()
-        dataset_type = request_json['datasetType']
-        if dataset_type not in Sample.DATASET_TYPE_LOOKUP:
-            raise ValueError('Invalid dataset type "{}"'.format(dataset_type))
+    required_fields = ['elasticsearchIndex', 'datasetType']
+    if any(field not in request_json for field in required_fields):
+        return create_json_response(
+            {'errors': ['request must contain fields: {}'.format(', '.join(required_fields))]}, status=400)
+
+    elasticsearch_index = request_json['elasticsearchIndex'].strip()
+    dataset_type = request_json['datasetType']
+    if dataset_type not in Sample.DATASET_TYPE_LOOKUP:
+        return create_json_response({'errors': ['Invalid dataset type "{}"'.format(dataset_type)]}, status=400)
 
+    try:
         sample_ids, sample_type = validate_index_metadata_and_get_elasticsearch_index_samples(
             elasticsearch_index, project=project, dataset_type=dataset_type)
         if not sample_ids:
             raise ValueError('No samples found in the index. Make sure the specified caller type is correct')
 
         sample_id_to_individual_id_mapping = load_mapping_file(
             request_json['mappingFilePath'], request.user) if request_json.get('mappingFilePath') else {}
-
-        loaded_date = timezone.now()
```

Comment:
none of this code changed, it just isn't indented anymore

Reply:
I see. And it also changes the exceptions to responses.